### ISSUE TITLE
Support bwa-meth alignment samples

### ISSUE THREAD

**Issue by vanreality** [CompEpigen/methylbert#6]
Hi, @hanyangii @lutsik, great work!

I have some bam files generated by bwa-meth and I wonder if methylbert has future update plan to support bwa-meth alignment.

To my knowledge, bwa-meth has higher mapping rate and similar accuracy compared to bismark under certain filtering conditions like `MAPQ > 10`. However, bwa-meth output bam file without `XM` tag.

I'm writing script to add `XM` tag to meet methylbert's need but still looking forward to hearing from you.

Best,

Fan

**Comment by hanyangii** [comment]
Hello, thanks a lot for your interest! 
Although I don't have first-hand experience with bwa-meth aligned BAM files, I am willing to update MethylBERT to support bwa-meth alignment in future. I just cannot guarantee when the update will be made. 

Meanwhile, if you want to generate the XM tag yourself, you can find the information here (p.9 - 12):
[https://www.bioinformatics.babraham.ac.uk/projects/bismark/Bismark_User_Guide.pdf](https://www.bioinformatics.babraham.ac.uk/projects/bismark/Bismark_User_Guide.pdf
)

Or, if possible, I would rather recommend to generate a processed data file yourself including 
"methyl_seq" column containing 0, 1, and 2 meaning unmethylated CpG, methylated CpG, and non-CpG. Please find the data format [03_Preprocessing_bulk_data.ipynb](https://github.com/CompEpigen/methylbert/blob/main/tutorials/03_Preprocessing_bulk_data.ipynb) in the tutorial. 
You don't need to make all of the columns. Only ["dna_seq", "methyl_seq", "ctype", "dmr_ctype", "dmr_label"] are needed in the file as shown in [dataset.py](https://github.com/CompEpigen/methylbert/blob/853285396921582f0696800c38755b48f0737459/src/methylbert/data/dataset.py#L30).

Hope my comment is helpful for you!

**Comment by vanreality** [comment]
Thanks a lot! It's very helpful.

**Comment by hanyangii** [comment]
Hello @vanreality 

I just looked at the bwa-meth aligner github page but could not find information about how read-level methylation patterns are stored in bwa-meth-aligned bam files. 

Could you please share it with me if you have any references/tutorials explaining read-level methylation patterns in bwa-meth?

Kind regards,
Yunhee

**Comment by vanreality** [comment]
I evaluated the callMethylation function in the epialleleR package, version 1.11.9, using a WGBS sample. By comparing the XM tag similarity of identical reads (the same read name and position) between bwameth and Bismark, I found that 98.61% of the reads in the bwameth BAM files matched perfectly with those in the Bismark BAM files.

Given these results, I plan to use this R package to extract read-level methylation information from BAM files aligned with bwameth.
